### ISSUE TITLE
Frequency of Specific  Number of  Dry Days Dialogue in R-Instat 

### ISSUE THREAD

**Issue by fran2or** [africanmathsinitiative/R-Instat#5919]
**Is your feature request related to a problem? Please describe.**
Work with some colleagues in Haiti have given rise to this problem.  In addition to having the feature of the longest dry spell which works well for them, they will also want a feature in R-Instat that can easily help them produce yearly  information of the frequency of a specific number of dry days within a period of every year available. For example the frequency of a 10-day dry spell in between August and October / on specific months?  

This could also lead to Markov chain idea as shown in the attached graphs.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
![image](https://user-images.githubusercontent.com/6714330/89561926-c08c9c80-d808-11ea-8f76-e040b47a5387.png)

**Comment by shadrackkibet** [comment]
@fran2or thanks for the feature request. A workaround is to use the **Climatic>Prepare>Climatic Summaries**. Top right option **Within Year** should be able to do this. Under summaries use **More Tab**(see below). You can change the threshold (7,10,15 etc) for the counts. Please make sure you rename a column after each summary is produced to avoid overriding the summary. Once you have your summaries then you can use **Describe>Specific>Line Plot** to obtain your desired graph. I think using **proportions** option will produce probabilities then you can use the same line plot dialog to obtain the graph. It is easy to add these options into the **Spells** dialogs where it belongs to make it easy to obtain such summaries. @rdstern?
![image](https://user-images.githubusercontent.com/26301973/89564831-4539e380-d826-11ea-94b8-3faa66138e32.png)
![image](https://user-images.githubusercontent.com/26301973/89566249-83380700-d828-11ea-89a6-3986a7592ea9.png)

**Comment by rdstern** [comment]
@fran2or very good idea - though I still think the results are difficult to use.  But as @shadrackkibet says, we should have them and it should be in the spells dialogue.  It is the parallel of the same idea in the extremes dialogue.  It should also be relatively easy to add.  I suggest our usual buttons, where the existing dialogue is the **_Maximum_** button and there is a second button with the label **_All_**.   

As with the extremes this is essentially a filter, adding a spells length variable (as in the transform dialogue) and chooses all rows where the next observation of the spell is zero, i.e. not in the spell.  As an option (like the extremes) you can choose only the spells longer than a given integer value.  Default is 0 to give all spells.  If there is a missing value in the data then the corresponding spell is also missing.

**Comment by dannyparsons** [comment]
I don't think this can quite be done in the way @shadrackkibet presents since it first requires calculating all the spells within the period and then a more complex filter. I like the filter method @rdstern mentions which would be simple to implement as an option to the spells dialogs.

@fran2or I think it can be done with a combination of dialogs in R-Instat currently. If you need to do this now in R-Instat let me know and I'll figure out the steps.

**Comment by dannyparsons** [comment]
Francis asked how to do this in the current version so here's what I did using the Ghana data in the library:

1. Climatic > Prepare > Transform. Choose Spells option, condition `<= 0.85` and check Calculate by year. This gives a column of spell count but we want to filter for only the end of the spell to get all the lengths. As Roger mentioned this is when the next spell value is 0.
2. Prepare > Calculator. `dplyr::lead(spell, 1)`. This will create another column with the lead value from the spell column i.e. the next value (opposite to a lag).
3. Filter. `spell != 0` and `spell_lead == 0`. `spell_lead == 0` gives us all the final lengths of the spells and `spell != 0` removes the `0` length spells (rainy days) that we are not interested in. Apply this filter as a subset so you can use it for other things. Now you have a data frame with all the spell lengths. You can do standard summaries to get the sort of calculations you mentioned.
4. e.g. I did a filter on the new data frame for `spell >= 7`. Then I did column summaries summarising by station and month and just doing the count summary (with no variables to summarise). Now I have a data frame with the number of spells with length greater than or equal to 7 per month per station. I could plot this with line plot dialog to get a similar plot to the one above.
